mne/time_frequency/psd.py

```diff
@@ -111,11 +101,11 @@ def _psd_welch(x, sfreq, fmin=0, fmax=np.inf, n_fft=256, n_overlap=0,
     x_splits = np.array_split(x, n_jobs)
     f_psd = parallel(my_pwelch(d, noverlap=n_overlap, nfft=n_fft,
                      fs=sfreq, freq_mask=freq_mask,
-                     welch_fun=welch)
+                     welch_fun=spectrogram)
                      for d in x_splits)
 
-    # Combining/reshaping to original data shape
-    psds = np.concatenate(f_psd, axis=0)
+    # Combining, reducing windows and reshaping to original data shape
+    psds = np.concatenate(f_psd, axis=0).mean(axis=-1)
```

Comment:
I suspect this will blow up memory as now you store all the FFTs before averaging.